packages/web3-eth-abi/src/index.js

```diff
@@ -268,8 +268,13 @@ ABICoder.prototype.formatParam = function (type, param) {
     if (match) {
         let size = parseInt(match[2] || "256");
         if (size / 8 < param.length) {
-            // pad to correct bit width
-            param = utils.leftPad(param, size);
+            if (param.startsWith('-')) {
+                // pad to correct bit width with - at the beginning
+                param = `-${utils.leftPad(param.substring(1), size)}`;
+            } else {
+                // pad to correct bit width
+                param = utils.leftPad(param, size);
+            }
```

Comment:
I based this branch off of #6238, so it includes all of those changes. The changes to this file are the only novel ones for this PR